src/coap_gnutls.c

```diff
@@ -1481,6 +1511,30 @@ coap_dtls_hello(coap_session_t *c_session,
   int ret;
 
   if (!g_env) {
+    /*
+     * Need to check that this actually is a Client Hello before wasting
+     * time allocating and then freeing off g_env.
+     */
+    if (data_len < sizeof(dtls_record_handshake_t)) {
+      coap_log(LOG_DEBUG,
+         "coap_dtls_hello: ContentType %d Short Packet (%ld < %ld) dropped\n",
+         data[OFF_CONTENT_TYPE], data_len, sizeof(dtls_record_handshake_t));
+      return 0;
+    }
+    if (data[OFF_CONTENT_TYPE] == DTLS_CT_HANDSHAKE &&
```

Comment:
Given that the specificity of log messages is not that important the condition could be rephrased as follows to combine this check with the check starting at line 1531.

```c
if ((data[OFF_CONTENT_TYPE] != DTLS_CT_HANDSHAKE)  ||
    (data[OFF_HANDSHAKE_TYPE] != DTLS_HT_CLIENT_HELLO)) {
```

Reply:
Updated as suggested (and replaced `sizeof(dtls_record_handshake_t)` with `(OFF_HANDSHAKE_TYPE + 1)` for consistency).